EMCAL/EMCALrec/AliEMCALClusterizerv3.h

```diff
@@ -0,0 +1,122 @@
+#ifndef ALIEMCALCLUSTERIZERV3_H
+#define ALIEMCALCLUSTERIZERV3_H
+
+/* Copyright(c) 1998-1999, ALICE Experiment at CERN, All rights reserved. *
+ * See cxx source for full Copyright notice                               */
+
+
+#include "AliEMCALClusterizerv1.h"
+class AliEMCALRecPoint; 
+class AliEMCALDigit;
+
+// Define numbers rows/columns for topological representation of cells
+const Int_t kNrows    = (24+1)*(6+4);  // 10x supermodule rows (6 for EMCAL, 4 for DCAL). +1 accounts for topological gap between two supermodules
+const Int_t kNcolumns = 48*2+1;        // 2x  supermodule columns + 1 empty space in between for DCAL (not used for EMCAL)
```

Comment:
The segmentation in phi (rows) is strictly speaking not correct because the small sectors only have 1/3 of the number of rows. Like this you allocate more than you actually need. And for run1 there is no DCAL which even further increases the unused memory. Also the gaps correspond to memory which is actually not used and maybe one can do the supermodule separation algorithmically.

Ideally the dimensions are taken from the AliEMCALGeometry, however in this case the arrays would have to be allocated dynamically.

Reply:
I think here we have to trade-off between speed and memory. The gaps are part of the algorithm by setting up a topology.

We should probably have in mind that the overall memory consumption of the topological maps is  (24+1) rows × (48×2 +1) columns ×(6+4) supermodules ×(1 Byte from cellmask+ 8 Bytes digitmap + (4+4+4) Bytes struct array) = 510 KB

I think we could as you say save memory, but on the cost of additional computations.
Getting the topological row/column while taking into account the small sectors requires at least "one subtraction operation" more per cell as far as I can see.
This could save us 4* (8 rows * 97 columns) * (1 Byte from cellmask+ 8 Bytes digitmap + (4+4+4) Bytes struct array)
= 65 KB

Leaving out the one-cell gaps would probably require many more computation operations. We would have to check the SM location algorithmically for each cell. This is naturally given with the one-cell gaps.
And here we are even talking about less memory to save:
(10 rows * 96 columns) + (250 rows * 1 column) * (1 Byte from cellmask+ 8 Bytes digitmap + (4+4+4) Bytes struct array) = 25 KB

So I think if we are not strongly memory-limited, the current approach is to be preferred in my opinion.